File: Monogame.Common/Camera.cs
using System;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace Monogame.Common
{
    public class Camera
    {
        private const float defaultZoom = 0.5f;
        
        private float zoom = defaultZoom;
        private float rotation = 0.0f;

        public Vector2 position = Vector2.Zero;

        public Matrix transform { get; private set; } = new Matrix();
        
        public Vector2 GetWorldPosition(Vector2 position2)
        {
            return Vector2.Transform(position2, Matrix.Invert(transform));
        }

        public Vector2 GetScreenPosition(Vector2 position2)
        {
            return Vector2.Transform(position2, transform);
        }

        public void Update(GraphicsDevice graphicsDevice)
        {
            var widthDiff = graphicsDevice.Viewport.Width / 1176.0f;
            var HeightDiff = graphicsDevice.Viewport.Height / 664.0f;

            transform =
                Matrix.CreateTranslation(
                    new Vector3(-position.X, -position.Y, 0.0f))
                        * Matrix.CreateRotationZ(rotation)
                        * Matrix.CreateScale(
                            new Vector3(
                                zoom * widthDiff
                                , zoom * HeightDiff
                                , 1.0f))
                        * Matrix.CreateTranslation(
                            new Vector3(
                                graphicsDevice.Viewport.Width * 0.5f
                                , graphicsDevice.Viewport.Height * 0.5f
                                , 0.0f));
        }
    }
}

File: AnimationTool/Camera.fs
module GameCamera
open Microsoft.Xna.Framework
open Microsoft.Xna.Framework.Graphics
open System

type Camera() = 
    let VIRTUAL_WIDTH = 1280.0f
    let VIRTUAL_HEIGHT = 720.0f
    let mutable Transform = Unchecked.defaultof<Matrix>
    let mutable Location = Vector2(50.0f,50.0f)
    member val Rotation = 0.0f with get, set
    member val Zoom = 5.0f with get, set

    member this.GetTransformation(graphicsDevice : GraphicsDevice) : Nullable<Matrix> =
        Transform <-
            Matrix.CreateTranslation(-Location.X, -Location.Y, 0.0f)
            * Matrix.CreateRotationZ(this.Rotation)
            * Matrix.CreateScale(
                this.Zoom * (float32 graphicsDevice.Viewport.Width / VIRTUAL_WIDTH)
                , this.Zoom * (float32 graphicsDevice.Viewport.Height / VIRTUAL_HEIGHT)
                , 1.0f)
            * Matrix.CreateTranslation(
                (float32 graphicsDevice.Viewport.Width) * 0.5f
                , (float32 graphicsDevice.Viewport.Height) * 0.5f
                , 0.0f)
        Nullable<Matrix> Transform

    member this.GetWorldPosition (position : Point) =
        this.GetWorldPosition(position.ToVector2())

    member this.GetWorldPosition (position : Vector2) =
        Vector2.Transform(
            position
            , Matrix.Invert(Transform)
        ).ToPoint()

    member this.GetScreenLocation(position : Vector2) =
        Vector2.Transform(position, Transform)